src/pydis_jam23/ui_app.py

```diff
@@ -0,0 +1,147 @@
+from PIL import Image
+from PyQt5 import QtGui, QtWidgets, uic
+
+from .codecs import CODECS, decode_message, encode_message
+
+# from .codecs.common import CodecError
+
+
+class UiApp(QtWidgets.QMainWindow):
+    def __init__(self):
+        super().__init__()
+
+        self.ui = uic.loadUi("src/pydis_jam23/ui_files/main.ui", self)
+        self.ui.setWindowIcon(QtGui.QIcon("src/pydis_jam23/ui_files/appicon.png"))  # we need a icon
+        # self.ui.setWindowTitle() # already set in 'main.ui' file but just incase
+        self._temp()
+        self._functions()
+        self.statusBar.showMessage("Application Started")
+        self.show()  # displays the QMainWindow in QApplication Instance
+
+    def _temp(self):
+        self._image_tmp = None
+        self.image = None
+
+    def _functions(self):
+        # initial widget state configs
+        self.lsbframe.hide()
+        self.lsbinsertdata.hide()
+
+        # shortcut key actions
+        self.action_open_image.setShortcut("Ctrl+O")
+        self.action_save_image.setShortcut("Ctrl+S")
+        self.action_exit_app.setShortcut("Ctrl+Q")
+
+        # lambda functions triggered by ui elements/buttons
+        self.action_open_image.triggered.connect(lambda: AppFuncs.action_open_image(self))
+        self.action_save_image.triggered.connect(lambda: AppFuncs.action_save_image(self))
+        self.action_exit_app.triggered.connect(lambda: AppFuncs.action_exit_app(self))
+        self.lsbtoolbtn.clicked.connect(lambda: AppFuncs.lsbtoolsection(self.statusBar, self.lsbframe))
+        self.lsbaction.currentTextChanged.connect(lambda: AppFuncs.lsbaction(self))
+        # self.lsbcodec.currentTextChanged.connect(lambda: AppFuncs.lsbsetcodec(self) )
+        self.lsbinsertbtn.clicked.connect(lambda: AppFuncs.lsbinsertdata(self))
+        self.lsbreceivebtn.clicked.connect(lambda: AppFuncs.lsbreceivedata(self))
+
+    # def resizeEvent(self, event):
+    # ^ this is a problem for hatch formatting because this method is inherited from QMainWindow
+    #     # self.image_scaled = image.scaled(self.scroll.width(),self.scroll.height())
+    #     # self.pixmap = QPixmap.fromImage(self.image_scaled)
+    #     # self.imageLabel.setPixmap(self.pixmap)
+    #     QtWidgets.QMainWindow.resizeEvent(self, event)
+    #     print(event)
+
+
+class AppFuncs:
+    @staticmethod
+    def action_open_image(root):
+        # this function opens a image in the gui app
+        root.statusBar.showMessage("file dialog opened")
+        filename = QtWidgets.QFileDialog.getOpenFileName()  # This is temporary
```

Comment:
This is temporary?

Reply:
This method allows any file to be opened.

Reply:
I'm just confused what the "This is temporary" comment is saying

Reply:
The method used to open a image file